core/services/keystore/eth.go

```diff
@@ -207,6 +178,72 @@ func (ks *eth) Export(id string, password string) ([]byte, error) {
 	return key.ToEncryptedJSON(password, ks.scryptParams)
 }
 
+// Get the next nonce for the given key and chain. It is safest to always to go the DB for this
+func (ks *eth) GetNextNonce(address common.Address, chainID *big.Int, qopts ...pg.QOpt) (int64, error) {
+	if !ks.exists(address) {
+		return 0, errors.Errorf("key with address %s does not exist", address.Hex())
+	}
+	return ks.orm.getNextNonce(address, chainID, qopts...)
+}
+
+// IncrementNextNonce increments keys.next_nonce by 1
+func (ks *eth) IncrementNextNonce(address common.Address, chainID *big.Int, currentNonce int64, qopts ...pg.QOpt) error {
+	if !ks.exists(address) {
+		return errors.Errorf("key with address %s does not exist", address.Hex())
+	}
+	return ks.orm.incrementNextNonce(address, chainID, currentNonce, qopts...)
+}
+
+func (ks *eth) Enable(address common.Address, chainID *big.Int, qopts ...pg.QOpt) error {
+	ks.lock.Lock()
+	defer ks.lock.Unlock()
+	return ks.enable(address, chainID, qopts...)
+}
+
+// caller must hold lock!
+func (ks *eth) enable(address common.Address, chainID *big.Int, qopts ...pg.QOpt) error {
```

Comment:
Note to review: please can you check I haven't messed up or missed any locking with these changes